api/test/integration/import-data/xlsx-uploads/import-data-produce-jobs.spec.ts

```diff
@@ -0,0 +1,58 @@
+import { Queue } from 'bull';
+import { Test, TestingModule } from '@nestjs/testing';
+import { AppModule } from 'app.module';
+import { ImportDataModule } from 'modules/import-data/import-data.module';
+import { getQueueToken } from '@nestjs/bull';
+import { importQueueName } from 'modules/import-data/workers/import-data.producer';
+import { ImportDataService } from 'modules/import-data/import-data.service';
+
+describe('XLSX Upload Feature Job Producer Tests', () => {
+  const importQueue: any = { add: jest.fn() };
+  const importQueueFail: any = { add: jest.fn().mockRejectedValue('test') };
+  let queue: Queue;
+  let moduleFixture: TestingModule;
+  let importDataService: ImportDataService;
+
+  const bootstrapTestingApp = async (fakeQueue: any): Promise<void> => {
+    moduleFixture = await Test.createTestingModule({
+      imports: [AppModule, ImportDataModule],
+    })
+      .overrideProvider(getQueueToken(importQueueName.value))
+      .useValue(fakeQueue)
+      .compile();
+
+    queue = moduleFixture.get(getQueueToken(importQueueName.value));
+    importDataService = moduleFixture.get(ImportDataService);
+  };
+
+  afterEach(async () => {
+    jest.clearAllMocks();
+    await moduleFixture.close();
+  });
+
+  test('When loadXlsxFile is called with required file data and a userId, Then a job should be pushed to the queue', async () => {
+    await bootstrapTestingApp(importQueue);
+    const fileData = { path: 'filepath' } as Express.Multer.File;
+    const userId = '29b47625-9ffa-45a9-bfce-7a75abeb820e';
+    await importDataService.loadXlsxFile(userId, fileData);
+
+    expect(queue.add).toHaveBeenCalledWith('excel-import-job', {
```

Comment:
This test leaves data in the queue. You have two options:
- either cleanup all queues on afterEach()
- Do something like https://github.com/Vizzuality/marxan-cloud/commit/0a87118dd45216ea6db3e8b5eee39fbb9a762057

Reply:
Event tho we are getting the queue by its token name after starting the module, what we are injecting as the queue instance before starting the test module is a mock object which is a just a jest spy, therefore this test does not add data to the queue. just triple checked locally again

What I had in mind and thought it would be a great task for @yulia-bel is setting up a test redis-db as we have with for the persistence layer so she can play a bit with redis. One of the reasons also why I created a different tasks for testing the events to let her play and get used to the queue events

Reply:
I missed the mock :facepalm: 

Not sure I understand where you want to set up a redis for testing. AFAIK we already have it on the servers, in CI and in the docker-compose for local execution, so not sure where it's missing. Let's chat about it, maybe

Reply:
Just a thought, even tho I have experience with queue stuff like activeMQ and SQS, I am new to redis. I saw it provides a way to use different 'databases' in the **same instance,** and also NestJS provides a way to pass different config values... so I thought we can already do it and that could be useful for some reason. I haven't think it deeply to be hones

Reply:
let's talk about this on Monday